packages/core/content-type-builder/server/src/controllers/validation/content-type.ts

```diff
@@ -140,7 +140,13 @@ const forbiddenContentTypeNameValidator = () => {
     name: 'forbiddenContentTypeName',
     message: `Content Type name cannot be one of ${reservedNames.join(', ')}`,
     test(value: unknown) {
-      return !(value && reservedNames.includes(value as string));
+      return !(
+        value &&
+        reservedNames.some((reserved) => {
+          // we compare snake case to check the actual column names that will be used in the database
+          return snakeCase(reserved) === snakeCase(value as string);
+        })
+      );
```

Comment:
Just a silly suggestion to get rid of the casting, and remove the negation on the whole condition (to make it easier to read). Feel free to discard it 

```suggestion
      if (typeof value !== 'string') {
        return false;
      }

      // we compare snake case to check the actual column names that will be used in the database
      return reservedNames.every((reservedName) => snakeCase(reservedName) !== snakeCase(value));
```